features/customers.feature

```diff
@@ -326,3 +326,23 @@ Scenario: Query a customer by city
     And I should see "Massachusetts" in the "State" field
     And I should see "USA" in the "Country" field
     And I should see "20000" in the "Zip" field
+
+Scenario: Update customer address
+    When I visit the "Home Page"
+    And I set the "Username" to "user2"
+    And I press the "Search for Customer" button
+
+    # Update the address
+    And I set the "Street Address" to "That Street"
+    And I set the "City" to "New Jersey"
+    And I set the "State" to "NJ"
+    And I set the "Country" to "America"
+    And I set the "Zip" to "11111"
+    And I press the "Update Customer Address" button
+    Then I should see the message "Success"
+    When I press the "Retrieve Address" button
```

Comment:
I'm wondering whether or not we should implement some step in web_steps.py to either copy and paste both the id's at the same time or to clear everything but the id numbers before we click retrieve address, just so the test shows that the id numbers are the only inputs supplied before retrieve address is pressed

Reply:
I think we should call clear_address_data() in rest_api.js when those buttons are clicked instead of in the web_steps.py

Reply:
I think though that with only that change, someone looking at the tests alone without looking at the details of the implementation wouldn't be able to tell that retrieve works with only the customer id and address id

Reply:
Yeah, but I'm thinking that even though we clear the address execpt the address_id, the users still cannot tell that retrieve works with only the customer id and address id. Because the form will be updated by the retrieved information from the server. I'm think maybe we can change the name of the retrieve buttons to 'retrieve by id'. If that helps to clearify this situation.

Reply:
Also, since we are using swagger now. I'm wondering that maybe we can replace the endpoint descriptions in the README.md with the UI instructions to explain what are those button doing, and how to use our UI.

Reply:
Yeah there's an issue for changing the README. I don't think adding the 'by id' to the retrieve button would help because the test still doesn't prove that the retrieve is by id only. I was just basing the copy/paste comment on what's done in lab-flask-bdd, but if you think we don't necessarily need to do that, we can stick with what we have

Reply:
Oh, yeah. I see what you mean. Working on pushing a new commit now